### DISCUSSION TITLE
How to compute a commitment using mimc

### DISCUSSION THREAD

**Discussion by jeljelih** [Consensys/gnark#905] [Q&A] [answered]
I would like to compute inside a gnark circuit the commitment of a witness message x with some witness randomness r and verify that it matches with the commitment (computed outside the circuit and passed as a public parameter).

I would like to check if my code corresponds to the correct way.

> mimc, _ := mimc.NewMiMC(api)
> mimc.Write(circuit.X)
> mimc.Write(circuit.Randomness)
> commitment := mimc.Sum()
> api.AssertIsEqual(circuit.ExpectedCommitment, commitment)

**Comment by ivokub** [comment, marked as answer]
This should work. Please note that MiMC hash is not universal as we have defined a MiMC instance per curve. So you would have to ensure that the witness you compute is computed using same MiMC instance as is used inside the circuit.

For debugging it would be better to have full minimal runnable program (including witness generation etc.)
